### ISSUE TITLE
INSTALL.rst confounds enduser installation with development instructions

### ISSUE THREAD

**Issue by hmaarrfk** [scikit-image/scikit-image#4672]
My big gripe with the INSTALL.rst file is that it doesn't distinguish two very distinct audiences:

1. Users that just want to *use* scikit-image.
2. Users that want to *hack* on scikit-image

The audience specified in 1 likely:
1. Likely doesn't have much experience with python.
2. Likely doesn't have much experience with isolated environments (and thus tried to use `sudo pip ...`
3. Likely already got to a seriously frustrating error before finding our installation instructions. Thus their mindset isn't the best.

The audience specified in 2 likely:
1. Already know how to setup a python build environment.
2. Likely just wants to know what dependencies to install since regular package management strategies are broken due to the need of a development build.
3. Is used to reading and adapting build instructions to their own environment.
4. Likely have already decided to invest more than 30 mins in this process.

For novices, some scary things I see just scanning through the document are:
* The starting assumption: `assuming that you have default Python environment already configured ` is very likely incorrect, as can be seen by many opened issues on this repo.
* Any mention of git (a scary technology that novices don't want to see)
* Any mention of formal testing (a scary technology that novices don't know about)

I strongly believe that the install instructions should be as simple as possible.
I'm happy to leave conda as a one liner, like what we do for Anaconda, and Python(x,y) and have the installation just focus on `pip`.

I think the instructions should really be as streamlined as possible, for the novice users.

Ideally, they should end with 
```
import skimage
print(skimage.__version__)
```
showing a very simple way for end users to know if they have installed the package.

Initial feedback of an attempt by developers has been given on PR
https://github.com/scikit-image/scikit-image/pull/4306

# References
Issues that would be better solved by:
> Please let us know if you've tried "step XXX" from the installation guide.

https://github.com/scikit-image/scikit-image/issues/4667
https://github.com/scikit-image/scikit-image/issues/4673

**Comment by alexdesiqueira** [comment]
Hey @hmaarrfk,

> I strongly believe that the install instructions should be as simple as possible.
> I'm happy to leave conda as a one liner, like what we do for Anaconda, and Python(x,y) and have the installation just focus on pip.

I disagree with you on the audience premises and the confusion between installation/dev install, but I strongly agree with you here. Maybe a "stable version/dev version" would suffice?

**Comment by hmaarrfk** [comment]
Definitely elaborate. I think the install instructions are outdated, and understanding the different usecases would be great.

**Comment by emmanuelle** [comment]
I also think that installation instructions should start with instructions for novices, therefore written in a simple way (not mentioning envs or git). Either we break this chapter in two, or we could also move the instruction for hackers to the contributing instructions (basically either you just want to use or you also want to modify the source code).

**Comment by alexdesiqueira** [comment]
> I think the install instructions are outdated, and understanding the different usecases would be great.

I don't understand the different use cases you cite here, but I'm all ears :)

> ... we could also move the instruction for hackers to the contributing instructions (basically either you just want to use or you also want to modify the source code).

I'd say, isn't the second option well treated on the contribution section?

**Comment by alexdesiqueira** [comment]
Maybe we could provide `requirements.txt`/`environment.yaml` files for installation of dependencies/environments as an alternative as well...?

**Comment by hmaarrfk** [comment]
ok, i think we are on the same page.

I'm not saying that the git instructions aren't worthwhile, but rather that the fact that `git ` appears in the install section in 2020 is really outdated.

This was very likely necessary in the days of poor packging support.

I'm not sure how to write install instructions with environments. How would you do it in a fool proof way?

```
pip install --user
```
breaks my conda environments ;), I just had to nuke `~/.local/python` or some directory like that.


I'm not too sure what do to do about conda. I use conda all the time, but even I would prefer omitting it from the instructions in favor of simplicity.

**Comment by hmaarrfk** [comment]
I think the install instructions are outdated.

Understanding the different use cases when we want to send user's to the install instructions is great.

**Comment by lagru** [comment]
I strongly agree with the core suggestion here. Thanks @hmaarrfk for making these points. A few thoughts:

- At least for me I expect an installation guide to address all relevant ways to install a package  including how to install from source. I'd be content with moving that to the contribution guide and link to it from the installation guide. That way it's as discoverable as before.
- What I don't expect to find inside the installation guide is a discussion on how to manage the dev dependencies (excluding build deps). Definitely agree with @hmaarrfk here.
- Beginners need instructions to be simple **but** without omission of critical info and context which prevent common pitfalls later on. The current version is definitely to terse for me in that regard and would profit at the very least from some context and specific links to related tool docs. 
- Installation of scikit-image and "proper" package management with environments are two separate things which both need to be explained (perhaps separately) to beginners. 
- I would assume that beginners either already have access to Python (and therefore at least to `python -m venv`) or already installed Anaconda in which case scikit-image is already included. In the latter case the only thing worth adding might be how to get the most recent version from conda-forge (like the current version already does but perhaps with more context).

From a cursory search it doesn't look like we get many support requests regarding the installation. Has this been a pain point?

**Comment by hmaarrfk** [comment]
There is likely 1 issue a week, somebody trying to "just get running" that can't install scikit-image.
The most recent one is:
https://github.com/scikit-image/scikit-image/issues/4721

Where the user likely needed to specify `--prefer-binary` flag.

**Comment by hmaarrfk** [comment]
@lagru please see the recent activity on indicating lack of clean instructions to get moving forward

https://github.com/scikit-image/scikit-image/issues/3649

which is indicating that users are having trouble dependint on tifffile due to old pip.

They are pinning to 0.14.2 instead of using the recently release 0.17.2 wheels that @emmanuelle worked so hard to get out.

**Comment by hmaarrfk** [comment]
Ok, i have my latest attempt here:

https://github.com/scikit-image/scikit-image/pull/4750

Where I discuss common pitfalls, after people tried "simple commands".

Where the developer instructions go, is still up for discussion. I just want to make sure we get he basics right.

**Comment by stefanv** [comment]
The README developer instructions are now out of date.

INSTALL.rst is outdated since we switced to Meson.

CONTRIBUTING.txt is most up to date.  Ensure we don't duplicate information between INSTALL and CONTRIBUTING?

**Comment by stefanv** [comment]
Since this is a developer-facing issue and we need to get the release out, I'm changing the milestone to 0.21.

**Comment by hmaarrfk** [comment]
I think we have done some improvements here. The `meson` build system is something I haven't kept up with admittedly which is why I have been silent on other issues where users have asked for help installing the system on windows.
